sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/NuGetMetadataReference.cs

```diff
@@ -109,110 +92,146 @@ private static MetadataReference[] CreateReferences(string packageId, string pac
 
         #endregion Helpers
 
-        private static Lazy<Dictionary<string, MetadataReference[]>> lazyFluentAssertions =
-            CreateLazy("FluentAssertions", "4.19.4", Constants.NuGetLatestVersion);
+        #region Package installation and update helpers
 
-        private static Lazy<Dictionary<string, MetadataReference[]>> lazyMicrosoftAspNetCoreMvcCore =
-            CreateLazy("Microsoft.AspNetCore.Mvc.Core", "2.0.4", Constants.NuGetLatestVersion);
+        private static void EnsurePackageIsInstalled(string packageId, string packageVersion)
+        {
+            if (packageVersion == Constants.NuGetLatestVersion)
+            {
+                if (IsCheckForLatestPackageRequired(packageId))
+                {
+                    LogMessage($"Checking for newer version of package: {packageId}");
+                    InstallPackage(packageId, packageVersion);
+                    WriteLastUpdateFile(packageId);
+                }
```

Comment:
It's probably worth also logging to say we skip because we already checked for a newer version recently.